commodore/compile.py

```diff
@@ -25,57 +26,57 @@
     ApiError,
     clean_working_tree,
     kapitan_compile,
-    lieutenant_query,
 )
 from .postprocess import postprocess_components
 from .refs import update_refs
+from .git import HEAD
 
 
 TARGET = "cluster"
 
 
-def _fetch_global_config(cfg, cluster):
-    config = cluster["base_config"]
+def fetch_global_config(cfg, cluster: Cluster):
     click.secho("Updating global config...", bold=True)
     repo = git.clone_repository(
-        f"{cfg.global_git_base}/{config}.git", "inventory/classes/global", cfg
+        cluster.global_git_repo_url(), "inventory/classes/global", cfg
     )
+    rev = cluster.global_git_repo_revision()
+    if rev != HEAD:
+        git.checkout_version(repo, rev)
```

Comment:
Consider getting rid of the constant `HEAD` and just return `None` when there's no revision overrides.

Then this could be simplified to

```suggestion
    if rev:
        git.checkout_version(repo, rev)
```